src/proc/rotation-transform.cpp

```diff
@@ -0,0 +1,86 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+
+#include "rotation-transform.h"
+
+#include "../include/librealsense2/hpp/rs_sensor.hpp"
+#include "../include/librealsense2/hpp/rs_processing.hpp"
+#include "context.h"
+#include "image.h"
+#include "stream.h"
+
+namespace librealsense
+{
+    rotation_transform::rotation_transform(rs2_format target_format)
+        : rotation_transform("Rotation Transform", target_format)
+    {}
+
+    rotation_transform::rotation_transform(const char* name, rs2_format target_format)
+        : stream_filter_processing_block(name)
+    {
+        _target_format = target_format;
+
+        if (_target_format == RS2_FORMAT_Z16)
+        {
+            _stream_filter.stream = RS2_STREAM_DEPTH;
+            _extension_type =RS2_EXTENSION_DEPTH_FRAME;
+        }
+        else if (_target_format == RS2_FORMAT_Y8)
+        {
+            _stream_filter.stream = RS2_STREAM_INFRARED;
+            _extension_type = RS2_EXTENSION_VIDEO_FRAME;
+        }
+        else if (_target_format == RS2_FORMAT_RAW8)
+        {
+            _stream_filter.stream = RS2_STREAM_CONFIDENCE;
+            _extension_type = RS2_EXTENSION_VIDEO_FRAME;
+        }
+        _stream_filter.format = target_format;
+    }
+
+    rs2::frame rotation_transform::process_frame(const rs2::frame_source& source, const rs2::frame& f)
+    {
+        auto p = f.get_profile();
+        if (p.get() != _source_stream_profile.get())
+        {
+            _source_stream_profile = p;
+            _target_stream_profile = p.clone(p.stream_type(), p.stream_index(), _target_format);
+
+            // Set the unique ID as the original frame.
+            // The frames are piped through a syncer and must have the origin UID.
+            auto target_spi = (stream_profile_interface*)_target_stream_profile.get()->profile;
+            target_spi->set_unique_id(p.unique_id());
```

Comment:
Is it possible to pass this condition as a flag to `_target_stream_profile = p.clone(p.s` ? so it will be performed during clone ?